prow/spyglass/podlogartifact.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/url"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/deck/jobs"
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+// PodLogArtifact holds data for reading from a specific pod log
+type PodLogArtifact struct {
+	name      string
+	buildID   string
+	podName   string
+	sizeLimit int64
+	ja        *jobs.JobAgent
```

Comment:
```
type podLogJobAgent interface {
  GetJobLog
  GetJobLogTail
}
```
and now you can easily unit test `PodLogArtifact`

Reply:
And the advantage to using this versus my current fake `JobAgent` is that this one gives me more control over `GetJobLog.*` implementations?

Reply:
Right. You need a way to inject errors into GetJobLog and GetJobLogTail. This is easier with a spy / fake implementation.